### PR TITLE
UI CDC: Show inserts, updates and deletes

### LINKED ISSUE #1520: UI Metrics: Segregate rows synced
Segregate the rows synced metrics into inserts, updates and deletes

### PR DESCRIPTION
**Pull request by Amogh-Bharadwaj**
This PR adds two visual elements to the CDC page in UI:
- Inserts, updates and deletes in total
- Inserts, updates and deletes for each table

Screenshot 2024-03-22 at 9 33 17 PM

Update: 
The I/U/D graph is now visible on click expand:
Screenshot 2024-03-25 at 8 31 38 PM

Fixes #1520

### CODE REVIEW

File: flow/connectors/kafka/kafka.go [hunk 1]
```diff
@@ -300,7 +301,14 @@ func (c *KafkaConnector) SyncRecords(ctx context.Context, req *model.SyncRecords
...
 				c.client.Produce(wgCtx, kr, produceCb)
-				tableNameRowsMapping[kr.Topic] += 1
+				switch record.(type) {
+				case *model.InsertRecord:
+					tableNameRowsMapping[kr.Topic].InsertCount += 1
+				case *model.UpdateRecord:
+					tableNameRowsMapping[kr.Topic].UpdateCount += 1
+				case *model.DeleteRecord:
+					tableNameRowsMapping[kr.Topic].DeleteCount += 1
+				}
```
Comment (serprex) [inline flow/connectors/kafka/kafka.go:311]: interface method on model.Record?

File: flow/connectors/utils/stream.go [hunk 2]
```diff
@@ -186,3 +187,16 @@ func recordToQRecordOrError(tableMapping map[string]uint32, batchID int64, recor
 		Record: entries[:],
 	}
 }
+
+func InitialiseTableRowsMap(tableMaps []*protos.TableMapping) map[string]*model.RecordTypeCounts {
+	tableNameRowsMapping := make(map[string]*model.RecordTypeCounts)
```
Comment (serprex) [inline flow/connectors/utils/stream.go:192]:
```suggestion
	tableNameRowsMapping := make(map[string]*model.RecordTypeCounts, len(tableMaps))
```

**Review by serprex — APPROVED** [review]

**Comment by Amogh-Bharadwaj** [comment]
> @Amogh-Bharadwaj lets put an info icon next to the "Rows Synced" and show this graph in a modal when that is pressed.

Can we have a chevron down arrow which when clicked, expands downward to show the graph ?

**Comment by iskakaushik** [comment]
> > @Amogh-Bharadwaj lets put an info icon next to the "Rows Synced" and show this graph in a modal when that is pressed.
> 
> Can we have a chevron down arrow which when clicked, expands downward to show the graph ?

sure

**Pull request merged**